File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int n = nums.size();
        if(n==1) return nums[0]+1;
        int sum = nums[0];
        int prev = nums[0];
        
        for(int i=1;i<n;i++){
            if(nums[i] == prev+1){
                sum += nums[i];
                prev = nums[i];
            }
            else break;
        }
        while(find(nums.begin(), nums.end(), sum) != nums.end()){
            sum++;
        }
        return sum;
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        n = len(nums)
        maxlen = 0
        if n==1:
            return nums[0]+1
        currsum =0
        mini =float('inf')
        currlength =0
        ptr=0
        indexmapped = [-1]*3000      
        for i in range(n):
            indexmapped[nums[i]] = i
        while ptr<n:
            if ptr > 0 and nums[ptr] != nums[ptr-1] + 1: 
                break
                
            currsum+=nums[ptr]
            currlength +=1
            if currlength>=maxlen:   
                maxlen = currlength
                res = currsum        
                while indexmapped[res]!=-1:
                    res+=1
                mini = res        
            ptr+=1
        return mini